### DISCUSSION TITLE
Disable form field with multiple conditions?

### DISCUSSION THREAD

**Discussion by yaplej** [filamentphp/filament#2601] [Help] [answered]
Hello again,

Is it possible to test for multiple conditions when disabling or hiding a field?

Something like this?
```
->disabled(fn(Component $livewire): bool {
                        if($livewire instanceof Pages\CreateModel){
                            return false;
                        }else{
                          // 2nd test.
                        }
                    })
```

**Comment by yaplej** [comment, marked as answer]
I think this should work but its not disabling the field on my edit page when that field has already been set.

```
->disabled(function (Component $livewire, ?Asset $record): Bool {
                        if($livewire instanceof Pages\EditAsset){
                            if($record->sn != null){
                                return true;
                            }
                        }

                        if($livewire instanceof Pages\CreateAsset){
                            return false;
                        }
                    })
```

**Reply by danharrin** [reply]
You're probably looking for `$get('sn')` rather than using `$record`, as `$record` only updates when the model is saved, but `$get()` can get the value from another field

https://filamentphp.com/docs/2.x/forms/advanced#using-closure-customisation

**Reply by yaplej** [reply]
Oh my goodness.  I was just tired.  That function worked just fine.  I was just checking the wrong field. :man_facepalming:
